stock_picking_auto_revert/models/stock_picking.py

```diff
@@ -0,0 +1,73 @@
+# Copyright 2019 ForgeFlow
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, models, exceptions, _
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    def _check_restrictions(self):
+        # to be implemented by other modules to add restrictions to returns
+        pass
+
+    @api.multi
+    def action_revert_recreate(self):
+        self.ensure_one()
+        pick = self
+        pick._check_restrictions()
+        msg = _(
+            "Too bad. This picking cannot be returned because the products "
+            "are not available in the destination location"
+        )
+        # Create return picking
+        StockReturnPicking = self.env["stock.return.picking"]
+        default_data = StockReturnPicking.with_context(
+            active_ids=pick.ids, active_id=pick.ids[0]
```

Comment:
Why not 
```suggestion
            active_ids=pick.ids, active_id=pick.id
```
as pick is protected by ensure_one()